### PR TITLE
Create mentoring.md

### PR DESCRIPTION
**Pull request by RomaLetodiani**

### CODE REVIEW

**Comment by RomaLetodiani** [comment]
anyone to review this PR?

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Mentoring
+
+## ETL
+
+### Problem asks
+Restructure grouped letters by point values into individual mappings.
```
Comment (IsaacG) [inline tracks/typescript/exercises/etl/etl/mentoring.md:6]:
```suggestion
### Problem asks

Restructure grouped letters by point values into individual mappings. 
```

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Mentoring
+
+## ETL
+
+### Problem asks
+Restructure grouped letters by point values into individual mappings. 
+Transform the old format (point-value groups)
```
Comment (IsaacG) [inline tracks/typescript/exercises/etl/etl/mentoring.md:7]: Exercism MD style is one line per sentence, one sentence per line.

**Review by IsaacG — COMMENTED** [review]
1. @SleeplessByte is the typescript track maintainer and probably the best to review this, assuming they have the availability for this. Mentor notes may not be a priority for them at the moment.
2. Mentor notes typically provide talking points and common gotchas or questions. Mentors can find community solutions pretty easily so finding a solution is pretty easy. It's the rest that provides much of the value. Can you provide notes for mentors beyond paraphrasing the exercise and providing a solution?

**Comment by RomaLetodiani** [comment]
@IsaacG 
Considered your suggestion and updated file

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+### Problem asks
+
+Restructure grouped letters by point values into individual mappings. 
+Transform the old format (point-value groups) into a new format (letter-score pairs) in TypeScript. 
+Associate lowercase letters with their corresponding point values, transitioning from grouped data to individual letter scoring.
+
+## Reasonable Solution
+
+The code snippet provided in TypeScript represents a function named transform.
```
Comment (SleeplessByte) [inline tracks/typescript/exercises/etl/etl/mentoring.md:13]:
```suggestion
The code snippet provided in TypeScript represents a function named `transform`. 
```

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+### Problem asks
+
+Restructure grouped letters by point values into individual mappings. 
+Transform the old format (point-value groups) into a new format (letter-score pairs) in TypeScript. 
+Associate lowercase letters with their corresponding point values, transitioning from grouped data to individual letter scoring.
+
+## Reasonable Solution
+
+The code snippet provided in TypeScript represents a function named transform. 
+This function takes in an object oldData, which has keys representing point values (as strings) and values as arrays of strings (letters associated with those point values).
```
Comment (SleeplessByte) [inline tracks/typescript/exercises/etl/etl/mentoring.md:14]:
```suggestion
This function takes in an object `oldData`, which has keys representing point values (as strings) and values as arrays of strings (letters associated with those point values).
```

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+The function iterates through the oldData object, assigning each letter (converted to lowercase) to its corresponding score in the newData object.
```
Comment (SleeplessByte) [inline tracks/typescript/exercises/etl/etl/mentoring.md:16]:
```suggestion
The function iterates through the `oldData` object, assigning each letter (converted to lowercase) to its corresponding score in the `newData` object. 
```

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+    for (const [score, letters] of Object.entries(oldData)) {
+        letters.forEach((letter: string) => {
+            newData[letter.toLowerCase()] = Number(score);
+        });
+    }
+
+    return newData;
+}
+```
```
Comment (SleeplessByte) [inline tracks/typescript/exercises/etl/etl/mentoring.md:62]: I don't know what this was supposed to be but it seems in the wrong place.

Comment (SleeplessByte) [inline tracks/typescript/exercises/etl/etl/mentoring.md:62]:
```suggestion
```

File: tracks/typescript/exercises/etl/etl/mentoring.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+- **ETL Concept:**
+  - Briefly explain the ETL concept, emphasizing its role in transforming data from one format to another.
+
+- **Type Safety:**
+  - Emphasize the use of TypeScript for type safety. Ensure that the input and output types are clearly defined to catch potential runtime errors.
+
+- **Functional Approach:**
+  - Introduce a more functional approach using methods like `map` and `reduce` for constructing the new data object, making the code more concise.
+
+This mentoring file aims to guide students through understanding the ETL problem, providing a reasonable solution, and offering suggestions for improvement and further exploration.
```
Comment (SleeplessByte) [inline tracks/typescript/exercises/etl/etl/mentoring.md:87]:
```suggestion
```

**Review by SleeplessByte — DISMISSED** [review]

**Review by SleeplessByte — APPROVED** [review]

**Comment by SleeplessByte** [comment]
Thank you very much. This is a good first version.

**Pull request merged**